src/hotspot/share/services/heapDumper.cpp

```diff
@@ -2027,52 +1987,41 @@ void DumperSupport::end_of_dump(AbstractDumpWriter* writer) {
   writer->write_u4(0);
 }
 
-// writes a HPROF_LOAD_CLASS record for the class (and each of its
-// array classes)
+// writes a HPROF_LOAD_CLASS record for the class
 void VM_HeapDumper::do_load_class(Klass* k) {
   static u4 class_serial_num = 0;
 
   // len of HPROF_LOAD_CLASS record
   u4 remaining = 2*oopSize + 2*sizeof(u4);
 
-  // write a HPROF_LOAD_CLASS for the class and each array class
-  do {
-    DumperSupport::write_header(writer(), HPROF_LOAD_CLASS, remaining);
-
-    // class serial number is just a number
-    writer()->write_u4(++class_serial_num);
+  DumperSupport::write_header(writer(), HPROF_LOAD_CLASS, remaining);
 
-    // class ID
-    Klass* klass = k;
-    writer()->write_classID(klass);
+  // class serial number is just a number
+  writer()->write_u4(++class_serial_num);
 
-    // add the Klass* and class serial number pair
-    dumper()->add_class_serial_number(klass, class_serial_num);
+  // class ID
+  Klass* klass = k;
```

Comment:
I don't think it's necessary to introduce the variable 'klass' since it's not a loop anymore.

Reply:
Fixed